features/ui/src/main/java/org/opennms/alec/rest/ALECRestImpl.java

```diff
@@ -1,26 +1,171 @@
 package org.opennms.alec.rest;
 
+import java.util.Arrays;
+import java.util.Optional;
+
 import javax.ws.rs.core.Response;
 
+import org.opennms.alec.data.DataStore;
+import org.opennms.alec.driver.main.Driver;
+import org.opennms.alec.engine.api.EngineFactory;
+import org.opennms.alec.engine.api.EngineRegistry;
+import org.opennms.alec.engine.cluster.ClusterEngineFactory;
+import org.opennms.alec.engine.dbscan.DBScanEngineFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 public class ALECRestImpl implements ALECRest {
     private static final Logger LOG = LoggerFactory.getLogger(ALECRestImpl.class);
+    public static final String ALEC_CONFIG = "ALEC_CONFIG";
+
+    private final DataStore dataStore;
+    private final BundleContext bundleContext;
+    private final ObjectMapper objectMapper;
+
+    public ALECRestImpl(DataStore dataStore, BundleContext bundleContext) {
+        this.dataStore = dataStore;
+        this.bundleContext = bundleContext;
+        objectMapper = new ObjectMapper();
+    }
 
     @Override
     public Response ping() {
-        // How do I identify the user?
+        return Response.ok("pong!!").build();
+    }
 
+    @Override
+    public Response getConfigurations() {
+        LOG.debug("Get Configurations");
+        ConfigurationImpl.Builder configuration = ConfigurationImpl.newBuilder();
+        KeyEnum.stream().forEach(keyEnum -> {
+            String value = (String) dataStore.get(keyEnum.toString(), ALEC_CONFIG).orElse("");
+            switch (keyEnum) {
+                case ENGINE:
+                    try {
+                        configuration.engineParameter(objectMapper.readValue(value, EngineParameter.class));
+                    } catch (JsonProcessingException e) {
+                        configuration.engineParameter(null);
+                    }
+                    break;
+                case AGREEMENT:
+                default:
+                    configuration.keyValue(KeyValueImpl.newBuilder().keyEnum(keyEnum).value(value).build());
+            }
+        });
+        return Response.ok().entity(configuration.build()).build();
+    }
 
+    @Override
+    public Response getConfiguration(KeyEnum keyEnum) {
+        LOG.debug("Get Configuration {}", keyEnum);
+        String value = (String) dataStore.get(keyEnum.toString(), ALEC_CONFIG).orElse("");
+        switch (keyEnum) {
+            case ENGINE:
```

Comment:
Can we remove this, in favour of just using the `/engine/configuration` instead?